cocinaborcelle/utils/userContext.tsx

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import { User } from "lucia";
+import { useRouter } from "next/router";
+import {
+	ReactNode,
+	createContext,
+	useContext,
+	useEffect,
+	useState,
+} from "react";
+
+const UserContext = createContext<User | null>(null);
+
+export function UserProvider({ children }: { children: ReactNode }) {
+	let authNeedPages = ["/user"];
```

Comment:
```suggestion

const pagesWithoutAuth = ["/", "/cadastro"];

export function UserProvider({ children }: { children: ReactNode }) {
```

é boa prática colocar dentro do componente React ou da página aquilo referente ao React. Essas definições de variáveis const é bom ficar fora do componente